packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -805,12 +818,21 @@ class _PageViewState extends State<PageView> {
 
     return NotificationListener<ScrollNotification>(
       onNotification: (ScrollNotification notification) {
-        if (notification.depth == 0 && widget.onPageChanged != null && notification is ScrollUpdateNotification) {
-          final PageMetrics metrics = notification.metrics as PageMetrics;
-          final int currentPage = metrics.page.round();
-          if (currentPage != _lastReportedPage) {
-            _lastReportedPage = currentPage;
-            widget.onPageChanged(currentPage);
+        if (notification.depth == 0) {
+          if (notification is ScrollUpdateNotification &&
```

Comment:
nit: keeping this condition here and the ones below in one line is probably more readable.